app/tests/conditional-layer/same-layer-reached-differently/native_posix.keymap

```diff
@@ -0,0 +1,69 @@
+#include <behaviors.dtsi>
+#include <dt-bindings/zmk/keys.h>
+#include <dt-bindings/zmk/kscan_mock.h>
+
+/ {
+    conditional_layers {
+        compatible = "zmk,conditional-layers";
+        conditional_layer_1 {
+            if-layers = <1 2>;
+            then-layer = <4>;
+        };
+        conditional_layer_2 {
+            if-layers = <1 3>;
+            then-layer = <4>;
+        };
+    };
+
+    keymap {
+        compatible = "zmk,keymap";
+        default_layer {
+            bindings = <
+                &kp A &mo 3
+                &mo 1 &mo 2
+            >;
+        };
+        layer_1 {
+            bindings = <
+                &kp B &trans
+                &trans &trans
+            >;
+        };
+        layer_2 {
+            bindings = <
+                &kp C &trans
+                &trans &trans
+            >;
+        };
+        layer_3 {
+            bindings = <
+                &kp D &trans
+                &trans &trans
+            >;
+        };
+        layer_4 {
+            bindings = <
+                &kp E &trans
+                &trans &trans
+            >;
+        };
+    };
+};
+
+&kscan {
```

Comment:
This behavior LGTM. Do you mind also adding a test case where all three `&mo` bindings are held at the same time, causing layer 4 to be activated through both the `<1 2>` and `<1 3>` configs?